src/main/scala/org/broadinstitute/dsde/agora/server/AgoraConfig.scala

```diff
@@ -67,6 +67,9 @@ object AgoraConfig {
   lazy val gcsUserEmail = config.as[String]("gcs.user.email")
   lazy val adminGoogleGroup = config.as[Option[String]]("admin.google.group")
 
+  // sam
+  lazy val samUrl = config.as[String]("sam.url")
+
```

Comment:
dumb question -- why do we tend to make these things lazy?  what's the advantage?

Reply:
I was just following the pattern of the rest of the code in this file... I don't think there is much advantage here.

But there might be 3 reasons: 1) don't spend the time to load it if you never need it, 2) spread out performance impact (i.e. not all of these things are loaded at object instantiation) and 3) complicated timing - the code is simpler to have a val in this scope but it is not ready to be evaluated until later.

Reply:
actually, if it were not lazy then the fiab tests would probably fail because firecloud-develop does not yet have the config changes on the dev branch

Reply:
i googled because i was interested and found [this](https://stackoverflow.com/questions/7484928/what-does-a-lazy-val-do) thread. tldr: what doge said, plus it can break cyclic dependencies